Svc/FpySequencer/FpySequencerStateMachine.fppi

```diff
@@ -0,0 +1,228 @@
+    @ iterates to the next statement and dispatches it
+    action dispatchStatement
+    @ resets the sequence runtime
+    action resetRuntime
+    @ checks if the current statement has timed out
+    action checkStatementTimeout
+    @ reports that the sequence timed out
+    action report_seqTimedOut
+
+    @ called in dispatchStatement method when a statement was successfully dispatched
+    signal result_dispatchStatement_success
+    @ called in dispatchStatement method when a statement was unable to be sent out
+    signal result_dispatchStatement_failure
+    @ called in dispatchStatement method when there were no more statements in the sequence
+    signal result_dispatchStatement_noMoreStatements
+    @ raised whenever the checkTimers port is called
+    signal checkTimersIn
+    @ raised when we are done sleeping
+    signal result_checkShouldWake_wakeup
+    @ raised when an operation could not be performed on a Fw::Time object due to a
+    @ mismatched time base or context
+    signal result_timeOpFailed
+    @ a directive is telling the sequencer to go to sleep
+    signal directiveResponse_beginSleep
+    @ a directive successfully executed
+    signal directiveResponse_success
+    @ a directive failed to execute
+    signal directiveResponse_failure
+    @ called when the expected cmd response comes in, and it is OK. only raised in the RUNNING.AWAITING_CMD_RESPONSE state
+    signal cmdResponse_success
+    @ called when the expected cmd response comes in, and it is not OK. only raised in the RUNNING.AWAITING_CMD_RESPONSE state
+    signal cmdResponse_failure
+    @ raised when the statement times out, according to the timeout parameter
+    signal statementTimeout
+
+    @ sequencer is executing all statements in the sequence
+    state RUNNING {
+        @ start with a fresh baked runtime
+        entry do { resetRuntime }
+
+        initial enter DISPATCH_STATEMENT
+
+        @ sequencer is stepping into a single statement and dispatching it
+        state DISPATCH_STATEMENT {
+            entry do { dispatchStatement }
+            # -->
+            on result_dispatchStatement_noMoreStatements do { report_seqSucceeded, sendCmdResponse_OK } enter IDLE
+            on result_dispatchStatement_failure do { report_seqFailed, sendCmdResponse_EXECUTION_ERROR } enter IDLE
+            on result_dispatchStatement_success enter AWAITING_STATEMENT_RESPONSE
+        }
+
+        state AWAITING_STATEMENT_RESPONSE {
+            on checkTimersIn do { checkStatementTimeout }
+            # -->
+            on statementTimeout do { report_seqTimedOut, sendCmdResponse_EXECUTION_ERROR } enter IDLE
+            on result_timeOpFailed do { report_seqFailed, sendCmdResponse_EXECUTION_ERROR } enter IDLE
+
+            on cmdResponse_success enter DISPATCH_STATEMENT
+            on cmdResponse_failure do { report_seqFailed, sendCmdResponse_EXECUTION_ERROR } enter IDLE
+
+            on directiveResponse_success enter DISPATCH_STATEMENT
```

Comment:
A comment here on the reason for replication between command and directive would help.